benchmarks/python/core.py

```diff
@@ -330,22 +356,59 @@ def run_benchmark(
         benchmark: pytest-benchmark fixture
         benchmark_fn: Target function
         inputs: Inputs to the target function
+        iobytes (Optional): When given, IO bytes computation is skipped
+                and this is used to compute the metrics.
+        device (Optional): Default: CUDA, Possible values: ["cuda", "host"].
+            Using device="host" is only allowed with nvFuser FusionDefinition.
+        fusion_fn (Optional): Must be provided if device = "host".
+            fusion_fn should only require FusionDefinition() as the input.
+            Use functools.partial if fusion_fn accepts additional arguments.
+            See test_many_pointwise_ops.py for example.
 
     Returns:
         outputs: Output of the target function
     """
 
+    assert device in [
+        "cuda",
+        "host",
+    ], f"Unsupported device type: {device}. Use one of cuda/host."
+
+    if device == "host":
+        assert fusion_fn is not None and benchmark_fn is None
+
+    nvf_benchmark = NVFBenchmark(benchmark, device=device)
+
+    def host_benchmark_fn(inputs, fd):
+        # Set the fd variable used to query the profile object
+        nvf_benchmark.fd = fd
+        return fd.execute(inputs, profile=True)
+
     def setup():
         clear_l2_cache()
+
+        if device == "host":
+            # Reset the FusionCache to measure host overhead correctly.
+            FusionCache.reset()
+            with FusionDefinition() as fd:
+                fusion_fn(fd)
```

Comment:
Since FusionCache.reset() needs to be done before the `with FusionDefinition()` block, the setup reinitializes fd at each round.

Reply:
Can you put that in a code comment?